### DISCUSSION TITLE
Conditional trigger fires on every insert.

### DISCUSSION THREAD

**Discussion by ttiras** [nhost/nhost#1148] [Q&A] [answered]
I have created below trigger on hasura console. Permissions to insert file for users are checked. 

```sql
CREATE OR REPLACE FUNCTION sync_student_files() RETURNS TRIGGER AS

$BODY$
    DECLARE session_variables JSON;
    DECLARE user_id uuid;

BEGIN
    session_variables := current_setting('hasura.user', 't')::JSON;
    user_id = (session_variables ->> 'x-hasura-user-id')::uuid;

    INSERT INTO
        public.user_files(user_id, file_id)
        VALUES(user_id,new.id);
        RETURN new;
END;
$BODY$
language plpgsql;

DROP TRIGGER IF EXISTS trig_sync_student_files ON storage.files;
CREATE TRIGGER trig_sync_student_files
     AFTER INSERT ON storage.files
     FOR EACH ROW
     WHEN (new.bucket_id = 'default')
     EXECUTE PROCEDURE sync_student_files();
```

When condition checks the new bucket_id and fires the trigger, but it is skipped somehow and the trigger is fired every time. I believe the syntax is correct but what else it can be?

**Comment by elitan** [comment, marked as answer]
This was fixed a few days ago in https://github.com/nhost/hasura-storage/pull/130 and released here: https://github.com/nhost/hasura-storage/releases/tag/v0.3.0. The version is available in production and in the last version of the CLI.

The issue was that we only inserted `id` to `storage.files` on the initial graphql insert for a new file. Since `bucket_id` is `default` as a default value, your trigger was always triggered.

Anyways, it should be fixed now.

**Comment by ttiras** [comment]
yes it works perfectly. thank you.
